goto.py

```diff
@@ -164,37 +177,38 @@ def _patch_code(code):
         target_depth = len(target_stack)
         if origin_stack[:target_depth] != target_stack:
             raise SyntaxError('Jump into different block')
+        
+        size = 0
+        for i in range(len(origin_stack) - target_depth):
+            size += _get_instruction_size('POP_BLOCK')
+        size += _get_instruction_size('JUMP_ABSOLUTE', target // _BYTECODE.jump_unit)
+        
+        moved_to_end = False
+        if pos + size > end:
+            # not enough space, add at end
+            pos = _write_instruction(buf, pos, 'JUMP_ABSOLUTE', len(buf) // _BYTECODE.jump_unit)
+            
+            if pos > end:
+                raise SyntaxError('Goto in an incredibly huge function') # not sure if reachable
+            
+            size += _get_instruction_size('JUMP_ABSOLUTE', end // _BYTECODE.jump_unit)
+            
+            moved_to_end = True
+            pos = len(buf)
+            buf.extend([0] * size)
 
-        failed = False
         try:
             for i in range(len(origin_stack) - target_depth):
                 pos = _write_instruction(buf, pos, 'POP_BLOCK')
-
-            if target >= end:
-                rel_target = (target - pos) // _BYTECODE.jump_unit
-                oparg_bits = 0
-
-                while True:
-                    rel_target -= (1 + _BYTECODE.argument.size) // _BYTECODE.jump_unit
-                    if rel_target >> oparg_bits == 0:
-                        pos = _write_instruction(buf, pos, 'EXTENDED_ARG', 0)
-                        break
-
-                    oparg_bits += _BYTECODE.argument_bits
-                    if rel_target >> oparg_bits == 0:
-                        break
-
-                pos = _write_instruction(buf, pos, 'JUMP_FORWARD', rel_target)
-            else:
-                pos = _write_instruction(buf, pos, 'JUMP_ABSOLUTE', target // _BYTECODE.jump_unit)
-
-        except (IndexError, struct.error):
-            failed = True
-
-        if failed or pos > end:
-            raise SyntaxError('Jump out of too many nested blocks')
-
-        _inject_nop_sled(buf, pos, end)
+            pos = _write_instruction(buf, pos, 'JUMP_ABSOLUTE', target // _BYTECODE.jump_unit)
+        except (IndexError, struct.error) as e:
+            raise SyntaxError("Internal error", e)
+            
+        if moved_to_end:
+            pos = _write_instruction(buf, pos, 'JUMP_ABSOLUTE', end // _BYTECODE.jump_unit)
+            
```

Comment:
Mind removing the blank line here?

Reply:
This blank line was indeed frivolous. 
Let me know if you want all other blank lines between if/elif to be removed in my latest pull request. Note that they make the functions more spaced out and readable in that case.

Reply:
I assumed this blank line was added by accident. In case it was not, I'd still make the point that we should keep things consistent, and so far we didn't add blank lines above `else` statements, and I'd prefer to keep it like that.